Framework/Parallel/CMakeLists.txt

```diff
@@ -0,0 +1,71 @@
+if ( MPI_EXPERIMENTAL )
+  set ( MPI_SRC_FILES
+    src/Communicator.cpp
+    src/ParallelRunner.cpp
+    src/Request.cpp
+    src/ThreadingBackend.cpp
+    )
+
+  set ( MPI_INC_FILES
+    inc/MantidParallel/Communicator.h
+    inc/MantidParallel/ParallelRunner.h
+    inc/MantidParallel/Request.h
+    inc/MantidParallel/ThreadingBackend.h
+    )
+
+  set ( MPI_TEST_FILES
+    CommunicatorTest.h
+    ParallelRunnerTest.h
+    RequestTest.h
+    ThreadingBackendTest.h
+    )
+endif ()
+
+set ( SRC_FILES
+	${MPI_SRC_FILES}
+	src/dummy.cpp
+)
+
+set ( INC_FILES
+	${MPI_INC_FILES}
+)
+
+set ( TEST_FILES
+	${MPI_TEST_FILES}
+)
+
+if (COVERALLS)
+  foreach( loop_var ${SRC_FILES} ${INC_FILES})
+    set_property(GLOBAL APPEND PROPERTY COVERAGE_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/${loop_var}")
+  endforeach(loop_var)
+endif()
+
+if(UNITY_BUILD)
```

Comment:
You can omit this, I think we need to go through and remove the remaining `UNITY_BUILD` macros and implementations at some point as they are unused.